### PR TITLE
add Closure support for fromEmail

### PR DESCRIPTION
**Pull request by bscheshirwork**
And set new email sender.

| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes/no (no one direct test for mailer)
| Fixed issues  | 271

### CODE REVIEW

**Comment by bscheshirwork** [comment]
Previous value make not sence , but some mailer (like gmail) replace it by account name

**Comment by maxxer** [comment]
What happens if the param `supportEmail` is not configured? I think Yii would throw an error. 

The basic app just have `adminEmail` by default.

**Comment by bscheshirwork** [comment]
> What happens if the param supportEmail is not configured? I think Yii would throw an error.
I can see this case in test log https://travis-ci.org/2amigos/yii2-usuario/jobs/442257715#L728
PHP error.

So... Will add aditional check for this. And yes... What is best way? Tell about set this property using throw error or set default module value? This value may be ignored by mail server. And mailer is in work state.

**Comment by maxxer** [comment]
I think allowing the closure is ok, but getting the value from params is not needed. Leave the default as is. 
The best you can do is explain how to change it in the installation instructions.

About Gmail yes, they do overwrite the `From` parameter with the authenticated username, so you cannot use their SMTP to send as another sender. Also that would likely break SPF and DKIM, so it would be useless.

**Comment by tonydspaniard** [comment]
@maxxer  @bscheshirwork  In my experience, even though Yii provides the great addition of closures on configuration arrays. I had problems using custom configuration builders due to the nature of closures it self. Thats why I try to avoid them on Yii as much as possible when about configuration is related.

**Comment by bscheshirwork** [comment]
@tonydspaniard this settings using on "bootstrap module" stage... This is simplest way for get a comiled config on this stage... IMHO...

File: docs/installation/mailer.md [hunk 1]
```diff
@@ -17,7 +17,23 @@ values:
 ]
 ```
 
-Actually, the only thing required is the `fromEmail` value. If you look at the code of `Da\User\Factory\MailFactory.php` 
+Actually, the only thing required is the `fromEmail` value. 
+May be we need to set this is equal to `supportEmail` from your `params` configs (like yii2-app-advanced template):
```
Comment (maxxer) [inline docs/installation/mailer.md]: I'd write:

*If you want to set it the same as supportEmail from your config params (like yii2-app-advanced template):*

Comment (maxxer) [reply docs/installation/mailer.md]: Engrish 🤣

**Pull request merged**